packages/rainbowkit/src/components/RainbowKitProvider/ModalContext.tsx

```diff
@@ -101,7 +101,10 @@ export function ModalProvider({ children }: ModalProviderProps) {
               ? openAccountModal
               : undefined,
           openChainModal:
-            connectionStatus === 'connected' ? openChainModal : undefined,
+            connectionStatus === 'connected' ||
+            connectionStatus === 'disconnected'
+              ? openChainModal
+              : undefined,
```

Comment:
To keep it simple we don't want to open chain modal if the connection status is either `loading` or `unauthenticated`. This is primarily used for our authentication provider.

- `loading` = fetching user
- `unauthenticated` = when the user has connected their wallet, but hasn't signed in

- `connected` = if user is logged in via siwe provider or has connected their wallet without using siwe auth provider
- `disconnected` = if user's wallet is disconnected